src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.CopyFile.OSX.cs

```diff
@@ -0,0 +1,111 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.IO
+{
+    internal static partial class FileSystem
+    {
+        public static partial void CopyFile(string sourceFullPath, string destFullPath, bool overwrite)
+        {
+            // Fail fast for blatantly copying onto self
+            if (sourceFullPath == destFullPath)
+            {
+                if (overwrite)
+                {
+                    throw new IOException(SR.Format(SR.IO_SharingViolation_File, destFullPath));
+                }
+
+                throw new IOException(SR.Format(SR.IO_FileExists_Name, destFullPath));
+            }
+
+            // Start by locking, creating relevant file handles, and reading out file status info.
+            using SafeFileHandle src = SafeFileHandle.OpenReadOnly(sourceFullPath, FileOptions.None, out var fileStatus);
+            UnixFileMode filePermissions = SafeFileHandle.GetFileMode(fileStatus);
+
+            // Read FileStatus of destination file to determine how to continue
+            int destError = Interop.Sys.Stat(destFullPath, out var destStat);
+
+            // Interpret the error from stat
+            if (destError != 0)
+            {
+                // Some error, let's see what it is:
+                Interop.ErrorInfo error = Interop.Sys.GetLastErrorInfo();
+
+                // Destination not existing is expected, so if this is the case we try clonefile,
+                // otherwise we got some other error that the fallback code can deal with.
+                if (error.Error == Interop.Error.ENOENT)
```

Comment:
```suggestion
                Interop.Error error = Interop.Sys.GetLastError();

                // Destination not existing is expected, so if this is the case we try clonefile,
                // otherwise we got some other error that the fallback code can deal with.
                if (error == Interop.Error.ENOENT)
```